config/config_test.go

```diff
@@ -50,12 +45,11 @@ func TestEnvOverrides_SwarmRedisPassword(t *testing.T) {
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			// Run test in non-concurrent way to avoid collisions of other test cases that parse config file
-			os.Args = tt.args
 			if tt.env != "" {
 				t.Setenv(redisPasswordEnv, tt.env)
 			}
-			err := cfg.Parse()
+			cfg := NewConfig()
+			err := cfg.ParseArgs(tt.args[0], tt.args[1:])
```

Comment:
```suggestion
			err := cfg.Parse(tt.args)
```

Reply:
Adding arguments to `cfg.Parse` function would cause API change which we would like to avoid to maintain backcompatibility.